@Dr15Jones the problem looks to be in this PR, while building the configuration:

```
Step: DQM Spec: ['@standardDQM', '@ExtraHLT', '@miniAODDQM', '@L1TEgamma']
replacing HLT process name - sequence DQMOffline will use 'reHLT'
replacing HLT process name - sequence DQMOfflineExtraHLT will use 'reHLT'
replacing HLT process name - sequence DQMOfflineMiniAOD will use 'reHLT'
replacing HLT process name - sequence DQMOfflineL1TEgamma will use 'reHLT'
customising the process with customisePostEra_Run2_2018 from Configuration/DataProcessing/RecoTLR
customising the process with customiseWithTimeMemoryJobReport from Validation/Performance/TimeMemoryJobReport
customising the process with miniAOD_customizeAllData from PhysicsTools/PatAlgos/slimming/miniAOD_tools
Traceback (most recent call last):
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/bin/slc7_amd64_gcc700/cmsDriver.py", line 56, in <module>
    run()
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/bin/slc7_amd64_gcc700/cmsDriver.py", line 28, in run
    configBuilder.prepare()
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/Configuration/Applications/ConfigBuilder.py", line 2270, in prepare
    self.pythonCfgCode += self.addCustomise(1)
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/Configuration/Applications/ConfigBuilder.py", line 886, in addCustomise
    self.process=getattr(package,fcn)(self.process)
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/PhysicsTools/PatAlgos/slimming/miniAOD_tools.py", line 538, in miniAOD_customizeAllData
    miniAOD_customizeCommon(process)
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/PhysicsTools/PatAlgos/slimming/miniAOD_tools.py", line 139, in miniAOD_customizeCommon
    applySubstructure( process )
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/PhysicsTools/PatAlgos/slimming/applySubstructure_cff.py", line 19, in applySubstructure
    addToProcessAndTask('ak4PFJetsPuppi'+postfix,ak4PFJetsPuppi.clone(), process, task)
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/PhysicsTools/PatAlgos/tools/helpers.py", line 30, in addToProcessAndTask
    setattr(process, label, module)
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/FWCore/ParameterSet/Config.py", line 444, in __setattr__
    self._replaceInSequences(name, newValue)
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/FWCore/ParameterSet/Config.py", line 924, in _replaceInSequences
    sequenceable._replaceIfHeldDirectly(old,new)
  File "/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_0_X_2019-07-10-2300/python/FWCore/ParameterSet/SequenceTypes.py", line 387, in _replaceIfHeldDirectly
    return self._seq._replaceIfHeldDirectly(original,replacement)
AttributeError: 'NoneType' object has no attribute '_replaceIfHeldDirectly'
```

--- a/FWCore/ParameterSet/python/Config.py
+++ b/FWCore/ParameterSet/python/Config.py
@@ -916,6 +916,12 @@ class Process(object):
     def _replaceInSequences(self, label, new):
         old = getattr(self,label)
         #TODO - replace by iterator concatenation
+        #to ovoid dependency problems between sequences, first modify
+        # process known sequences to do a non-recursive change. Then do
+        # a recursive change to get cases where a sub-sequence unknown to
+        # the process has the item to be replaced
+        for sequenceable in six.itervalues(self.sequences):
+            sequenceable._replaceIfHeldDirectly(old,new)
         for sequenceable in six.itervalues(self.sequences):
             sequenceable.replace(old,new)
         for sequenceable in six.itervalues(self.paths):
@@ -2007,6 +2013,7 @@ process.s2 = cms.Sequence(process.a+(process.a+process.a))""")
             s.associate(t2)
             p.s4.associate(t2)
             p.p = Path(p.c+s+p.a)
+            p.p2 = Path(p.c+p.s4+p.a)
             p.e3 = EndPath(p.c+s+p.a)
             new = EDAnalyzer("NewAnalyzer")
             new2 = EDProducer("NewProducer")
@@ -2019,12 +2026,18 @@ process.s2 = cms.Sequence(process.a+(process.a+process.a))""")
             visitor2 = NodeVisitor()
             p.p.visit(visitor2)
             self.assertTrue(visitor2.modules == set([new,new2,p.b,p.c]))
+            self.assertEqual(p.p.dumpPython()[:-1], "cms.Path(process.c+process.a+process.b+process.a, cms.Task(process.d))")
+            visitor_p2 = NodeVisitor()
+            p.p2.visit(visitor_p2)
+            self.assertTrue(visitor_p2.modules == set([new,new2,p.b,p.c]))
+            self.assertEqual(p.p2.dumpPython()[:-1], "cms.Path(process.c+process.s4+process.a)")
             visitor3 = NodeVisitor()
             p.e3.visit(visitor3)
             self.assertTrue(visitor3.modules == set([new,new2,p.b,p.c]))
             visitor4 = NodeVisitor()
             p.s4.visit(visitor4)
             self.assertTrue(visitor4.modules == set([new,new2,p.b]))
+            self.assertEqual(p.s4.dumpPython()[:-1],"cms.Sequence(process.a+process.b, cms.Task(process.d))")
             visitor5 = NodeVisitor()
             p.t1.visit(visitor5)
             self.assertTrue(visitor5.modules == set([new2]))

--- a/FWCore/ParameterSet/python/SequenceTypes.py
+++ b/FWCore/ParameterSet/python/SequenceTypes.py
@@ -184,6 +184,20 @@ class _SequenceCollection(_Sequenceable):
         return self._collection.index(item)
     def insert(self,index,item):
         self._collection.insert(index,item)
+    def _replaceIfHeldDirectly(self,original,replacement):
+        didReplace = False
+        for i in self._collection:
+            if original == i:
+                self._collection[self._collection.index(original)] = replacement
+                didReplace = True
+            elif isinstance(i,_UnarySequenceOperator):
+                if i._replace(original, replacement):
+                    didReplace = True
+                    if replacement is None:
+                        self._collection[self._collection.index(i)] = None
+        if replacement is None:
+            self._collection = [ i for i in self._collection if i is not None]
+        return didReplace
 
 
 
@@ -364,6 +378,20 @@ class _ModuleSequenceType(_ConfigureComponent, _Labelable):
             self._tasks.clear()
             self.associate(*v.result(self)[1])
         return v.didReplace()
+    def _replaceIfHeldDirectly(self,original,replacement):
+        """Only replaces an 'original' with 'replacement' if 'original' is directly held.
+            If another Sequence or Task holds 'original' it will not be replaced."""
+        didReplace = False
+        if original in self._tasks:
+            self._tasks.remove(original)
+            if replacement is not None:
+                self._tasks.add(replacement)
+            didReplace = True
+        if self._seq is not None:
+            didReplace |= self._seq._replaceIfHeldDirectly(original,replacement)
+        return didReplace
+    
+    
     def index(self,item):
         """Returns the index at which the item is found or raises an exception"""
         if self._seq is not None:
@@ -453,8 +481,8 @@ class _UnarySequenceOperator(_BooleanLogicSequenceable):
     def _replace(self, original, replacement):
         if self._operand == original:
             self._operand = replacement
-        else:
-            self._operand._replace(original, replacement)
+            return True
+        return False
     def _remove(self, original):
         if (self._operand == original): return (None, True)
         (self._operand, found) = self._operand._remove(original)
@@ -1881,6 +1909,46 @@ if __name__=="__main__":
             t3 = Task(m5)
             t2.replace(m2,t3)
             self.assertTrue(t2.dumpPython(None) == "cms.Task(process.m1, process.m3, process.m5)\n")
+
+        def testReplaceIfHeldDirectly(self):
+            m1 = DummyModule("m1")
+            m2 = DummyModule("m2")
+            m3 = DummyModule("m3")
+            m4 = DummyModule("m4")
+            m5 = DummyModule("m5")
+            
+            s1 = Sequence(m1*~m2*m1*m2*ignore(m2))
+            s1._replaceIfHeldDirectly(m2,m3)
+            self.assertEqual(s1.dumpPython()[:-1],
+                             "cms.Sequence(process.m1+~process.m3+process.m1+process.m3+cms.ignore(process.m3))")
+
+            s2 = Sequence(m1*m2)
+            l = []
+            s3 = Sequence(~m1*s2)
+            s3._replaceIfHeldDirectly(~m1, m2)
+            self.assertEqual(s3.dumpPython()[:-1],
+                             "cms.Sequence(process.m2+(process.m1+process.m2))")
+
+            m6 = DummyModule("m6")
+            m7 = DummyModule("m7")
+            m8 = DummyModule("m8")
+            m9 = DummyModule("m9")
+            t6 = Task(m6)
+            t7 = Task(m7)
+            t89 = Task(m8, m9)
+            
+            s1 = Sequence(m1+m2, t6)
+            s2 = Sequence(m3+m4, t7)
+            s3 = Sequence(s1+s2, t89)
+            s3._replaceIfHeldDirectly(m3,m5)
+            self.assertEqual(s3.dumpPython()[:-1], "cms.Sequence(cms.Sequence(process.m1+process.m2, cms.Task(process.m6))+cms.Sequence(process.m3+process.m4, cms.Task(process.m7)), cms.Task(process.m8, process.m9))")
+            s2._replaceIfHeldDirectly(m3,m5)
+            self.assertEqual(s2.dumpPython()[:-1],"cms.Sequence(process.m5+process.m4, cms.Task(process.m7))")
+            self.assertEqual(s3.dumpPython()[:-1], "cms.Sequence(cms.Sequence(process.m1+process.m2, cms.Task(process.m6))+cms.Sequence(process.m5+process.m4, cms.Task(process.m7)), cms.Task(process.m8, process.m9))")
+        
+            s1 = Sequence(t6)
+            s1._replaceIfHeldDirectly(t6,t7)
+            self.assertEqual(s1.dumpPython()[:-1],"cms.Sequence(cms.Task(process.m7))")
 
         def testIndex(self):
             m1 = DummyModule("a")

